src/nnet/nnet-parametric-relu.h

```diff
@@ -0,0 +1,284 @@
+      "\n  min_beta" + MomentStatistics(min_beta_) +
+      ", min-lr-coef " + ToString(min_learn_rate_coef_);
+  }
+  std::string InfoGradient() const {
+    return std::string("\n  max_alpha_grad") +
+      MomentStatistics(max_alpha_corr_) +
+      ", max-lr-coef " + ToString(max_learn_rate_coef_) +
+      "\n  min_beta_grad" + MomentStatistics(min_beta_corr_) +
+      ", min-lr-coef " + ToString(min_learn_rate_coef_);
+  }
+
+  void PropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                    CuMatrixBase<BaseFloat> *out) {
+    out->ParametricRelu(in, max_alpha_, min_beta_);
+  }
+
+  void BackpropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                        const CuMatrixBase<BaseFloat> &out,
+                        const CuMatrixBase<BaseFloat> &out_diff,
+                        CuMatrixBase<BaseFloat> *in_diff) {
+    // multiply error derivative by activations (alpha, beta)
+    // ((dE/da)*w) === out_diff, f(y) == out,
+    //  (out > 0 ) ? out_diff * alpha : out_diff * beta
+    in_diff->DiffParametricRelu(out, out_diff, max_alpha_, min_beta_);
+  }
+
+
+  void Update(const CuMatrixBase<BaseFloat> &input,
+              const CuMatrixBase<BaseFloat> &diff) {
+    // we use following hyperparameters from the option class
+    const BaseFloat alr = opts_.learn_rate * max_learn_rate_coef_;
+    const BaseFloat blr = opts_.learn_rate * min_learn_rate_coef_;
+    const BaseFloat mmt = opts_.momentum;
+    // compute gradient (incl. momentum)
+    if (!max_fix_coef_) {  // the alpha parameter is learnable
+       input_max_.Resize(input.NumRows(), input.NumCols());
+       input_max_.CopyFromMat(input);
+       input_max_.ApplyFloor(0.0);
+       max_alpha_corr_.AddRowSumMat(1.0, input_max_, mmt);
+       max_alpha_.AddVec(-alr, max_alpha_corr_);
+    }
+    if (!min_fix_coef_) {  // the beta parameter is learnable
+       input_min_.Resize(input.NumRows(), input.NumCols());
+       input_min_.CopyFromMat(input);
+       input_min_.ApplyCeiling(0.0);
+       min_beta_corr_.AddRowSumMat(1.0, input_min_, mmt);
+       min_beta_.AddVec(-blr, min_beta_corr_);
+    }
+    if (clip_gradient_ > 0.0) {  // gradient clipping
+      max_alpha_corr_.ApplyFloor(-clip_gradient_);
+      min_beta_corr_.ApplyFloor(-clip_gradient_);
+      max_alpha_corr_.ApplyCeiling(clip_gradient_);
+      min_beta_corr_.ApplyCeiling(clip_gradient_);
+    }
+  }
+
+  /// Accessors to the component parameters,
+  const CuVectorBase<BaseFloat>& GetAlpha() const { return max_alpha_; }
+
+  void SetAlpha(const CuVectorBase<BaseFloat>& max_alpha) {
+    KALDI_ASSERT(max_alpha.Dim() == max_alpha_.Dim());
+    max_alpha_.CopyFromVec(max_alpha);
+  }
+  const CuVectorBase<BaseFloat>& GetBeta() const { return min_beta_; }
+
+  void SetBeta(const CuVectorBase<BaseFloat>& min_beta) {
+    KALDI_ASSERT(min_beta.Dim() == min_beta_.Dim());
+    min_beta_.CopyFromVec(min_beta);
+  }
+
+ private:
+  CuVector<BaseFloat> max_alpha_;
```

Comment:
I would be good to write the purpose of the variables in a comment... It can look like this:
BaseFloat max_alpha_;  ///< Vector of 'alphas', one value per neuron.